src/reflect/scala/reflect/internal/util/Collections.scala

```diff
@@ -201,8 +201,10 @@ trait Collections {
   final def mapFrom[A, A1 >: A, B](xs: List[A])(f: A => B): Map[A1, B] = {
     Map[A1, B](xs map (x => (x, f(x))): _*)
   }
-  final def linkedMapFrom[A, A1 >: A, B](xs: List[A])(f: A => B): mutable.LinkedHashMap[A1, B] = {
-    mutable.LinkedHashMap[A1, B](xs map (x => (x, f(x))): _*)
+  final def linkedMapFrom[A, A1 >: A, B](xs: Iterator[A])(f: A => B): mutable.LinkedHashMap[A1, B] = {
+    val res = mutable.LinkedHashMap.empty[A1, B]
+    xs.foreach(x => res +=  x -> f(x) )
```

Comment:
Might as well commit to `while (xs.hasNext) { val x = xs.next(); res += x -> f(x) }`